apps/site/test/site_web/plugs/year_and_month_test.exs

```diff
@@ -0,0 +1,63 @@
+defmodule SiteWeb.Plugs.YearMonthTest do
+  @moduledoc false
+  use SiteWeb.ConnCase, async: true
+  import SiteWeb.Plugs.YearMonth
+
+  @assigned_date ~D[2020-07-17]
+
+  describe "call/2" do
+    setup %{conn: conn} do
+      conn = conn |> assign(:date, @assigned_date)
+      {:ok, conn: conn}
+    end
+
+    test "basic assigns", %{conn: conn} do
+      conn =
+        conn
+        |> with_query_params(%{
+          "year" => "2019",
+          "month" => "4"
+        })
+        |> call([])
+
+      assert %{year: 2019, month: 4} = conn.assigns
+    end
+
+    test "defaults to current", %{conn: conn} do
+      conn = conn |> call([])
+      %{year: year, month: month} = conn.assigns
+      assert ^year = @assigned_date.year
+      assert ^month = @assigned_date.month
+    end
+
+    test "defaults if invalid params", %{conn: conn} do
+      conn =
+        conn
+        |> with_query_params(%{
+          "year" => "2019",
+          "month" => "40"
+        })
+        |> call([])
+
+      %{year: year, month: month} = conn.assigns
+      assert 2019 = year
```

Comment:
Is this `=` or `==`? Or just Elixir pattern matching? 😅

Reply:
Haha just pattern matching!